src/server/req_delete.c

```diff
@@ -595,13 +720,8 @@ req_deletejob(struct batch_request *preq)
 							   preq->rq_host);
 						job_purge(pjob);
 					} else {
-						if (dup_br_for_subjob(preq, pjob, req_deletejob2) == 2) {
-							preq->rq_ind.rq_deletejoblist.jobid_to_resume = j;
-							preq->rq_ind.rq_deletejoblist.subjobid_to_resume = i;
-							--preq->rq_refct;
-							preply->brp_un.brp_deletejoblist.tot_arr_jobs--;
-							return;
-						}
+						if (dup_br_for_subjob(preq, pjob, req_deletejob2) == 2)
+							continue;
```

Comment:
don't we need to return from here? ==2 means that we hit a transit state job, for which we need to pause deljoblist, right?

Reply:
No delete list can continue. The subjob has its on request structure which will work on its own.